packages/api/src/rich-text/detection.ts

```diff
@@ -0,0 +1,91 @@
+import TLDs from 'tlds'
+import { AtpAgent } from '../agent'
+import { AppBskyRichtextFacet } from '../client'
+import { UnicodeString } from './unicode'
+
+export type Facet = AppBskyRichtextFacet.Main
+
+export async function detectFacets(
+  agent: AtpAgent,
+  text: UnicodeString,
+): Promise<Facet[] | undefined> {
+  let match
+  const facets: Facet[] = []
+  {
+    // mentions
+    const re = /(^|\s|\()(@)([a-zA-Z0-9.-]+)(\b)/g
```

Comment:
we actually have a handle regex now over in the identifiers package if you wanna use that 👀

Reply:
oh it doesn't take into account the leading `@` tho